velox/functions/prestosql/ArrayContains.cpp

```diff
@@ -47,9 +48,17 @@ void applyTyped(
       auto offset = rawOffsets[indices[row]];
 
       for (auto i = 0; i < size; i++) {
-        if (rawElements[offset + i] == search) {
-          flatResult.set(row, true);
-          return;
+        if constexpr (std::is_floating_point_v<T>) {
```

Comment:
Should we handle nan at line 79 too?

Also, do we need to handle nan inside complex element types?

Reply:
Good catch, fixed it and added additional tests for this and complex elements! Thanks!